File: tests/test_complete_cmd.py
from tests.utils import CLITestCase
from task_forge.sdk.types import Task
from task_forge.cli.complete_cmd import run


class TestComplete(CLITestCase):
    def test_complete_cmd(self):
        task = Task(id="test", title="test", priority=1)
        args = self.args()
        args["<ID>"] = [task.id]
        run(args)
        self.client.tasks.complete_by_id.assert_called_with(task.id)

    def test_completes_current_with_no_ids(self):
        task = Task(id="test", title="test", priority=1)
        self.client.tasks.current.return_value = task
        args = self.args()
        run(args)
        self.client.tasks.complete_by_id.assert_called_with(task.id)

File: tests/test-complete-cmd.ts
import { cli, listTasks, generateTask, generateUser } from './utils';

describe('task complete', () => {
    test('no arguments completes the current task', async () => {
        const { token } = await generateUser();
        await generateTask(token);
        const { id } = await generateTask(token, { priority: 2 });
        await cli('complete', token);
        const tasks = await listTasks(token);
        for (const task of tasks) {
            if (task.id === id) {
                expect(task.completed_date).not.toBe(null);
            } else {
                expect(task.completed_date).toBe(null);
            }
        }
    });

    test('given an id completes the indicated task', async () => {
        const { token } = await generateUser();
        const { id } = await generateTask(token);
        await generateTask(token, { priority: 2 });
        await cli(`complete ${id}`, token);
        const tasks = await listTasks(token);
        for (const task of tasks) {
            if (task.id === id) {
                expect(task.completed_date).not.toBe(null);
            } else {
                expect(task.completed_date).toBe(null);
            }
        }
    });

    test('accepts multiple ids and completes the indicated tasks', async () => {
        const { token } = await generateUser();
        const { id } = await generateTask(token);
        const task2 = await generateTask(token, { priority: 2 });
        await cli(`complete ${id} ${task2.id}`, token);
        const tasks = await listTasks(token);
        for (const task of tasks) {
            expect(task.completed_date).not.toBe(null);
        }
    });

    test('done is an alias to complete', async () => {
        const { token } = await generateUser();
        await generateTask(token);
        const { id } = await generateTask(token, { priority: 2 });
        await cli('done', token);
        const tasks = await listTasks(token);
        for (const task of tasks) {
            if (task.id === id) {
                expect(task.completed_date).not.toBe(null);
            } else {
                expect(task.completed_date).toBe(null);
            }
        }
    });
});
